src/main/java/net/minestom/server/network/NetworkBufferImpl.java

```diff
@@ -555,6 +554,35 @@ static NetworkBufferImpl impl(NetworkBuffer buffer) {
         return (NetworkBufferImpl) buffer;
     }
 
+    BinaryTagWriter nbtWriter() {
+        if (this.nbtWriter == null) {
+            this.nbtWriter = new BinaryTagWriter(new DataOutputStream(new OutputStream() {
+                @Override
+                public void write(int b) {
+                    NetworkBufferImpl.this.write(BYTE, (byte) b);
+                }
+            }));
+        }
+        return this.nbtWriter;
+    }
+
+    BinaryTagReader nbtReader() {
+        if (nbtReader == null) {
+            this.nbtReader = new BinaryTagReader(new DataInputStream(new InputStream() {
+                @Override
+                public int read() {
+                    return NetworkBufferImpl.this.read(BYTE) & 0xFF;
+                }
+
+                @Override
+                public int available() {
+                    return (int) NetworkBufferImpl.this.readableBytes();
+                }
+            }));
+        }
+        return nbtReader;
```

Comment:
Looks dirty

Reply:
Non-actionable feedback, it's the best way to my knowledge of doing this.